lib/ex_doc/formatter/html/templates.ex

```diff
@@ -90,6 +100,13 @@ defmodule ExDoc.Formatter.HTML.Templates do
         group: to_string(group),
         headers: extract_headers(content)
       }
+      |> then(fn item ->
+        if is_nil(extra[:search_data]) do
+          item
+        else
+          Map.put(item, :hide_in_autocomplete, true)
+        end
+      end)
```

Comment:
I'd like to suggest one alternative approach here. Instead of introducing `custom` as its own separate thing, make this be:

    Map.put(item, :custom, [%{id: ..., anchor: ...}, ...])

Where the entry in the list is as closest to the header as possible, but feel free to add additional fields if you find necessary. I think this will require fewer changes to the autocomplete code and be similar in nature.

The only caveat is the empty anchor. You would either have to remove it here, since we would still automatically autocomplete the top/parent node, or we take the decision that search_data is only about the children data. The top/parent node is always automatically included, both in autocomplete, and in search.

Reply:
Sorry, I’m not following 😓

Reply:
To recap: instead of storing a separate `nodes.custom`, we will have an optional `.search` field inside each extra. Then the autocomplete code will do this: if the element has a `.search` field, you will search for entries inside the .search field, otherwise business as usual. This means that the results will still appear under the "extras" category instead of adding a new one. Also, the entries in the .search field should mirror the ones in `headers` as much as possible for simplicity. And then we can also get rid of `hide_in_autocomplete`.